### PR TITLE
Wip/hughsie/private serial number

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
Does this really need a new flag?  Can't you just go directly off of whether this is an internal to make a decision?

File: libfwupdplugin/fu-device.h [hunk 1]
```diff
@@ -447,6 +447,18 @@ typedef guint64 FuDeviceInternalFlags;
  */
 #define FU_DEVICE_INTERNAL_FLAG_MD_SET_SIGNED (1ull << 23)
 
+/**
+ * FU_DEVICE_INTERNAL_FLAG_PRIVATE_SERIAL_NUMBER:
+ *
+ * Keep the device serial number and instance IDs private for non-root users.
+ *
+ * NOTE: This doesn't make sense for removable devices as the attacker could just unplug the device
```
[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-device.h:455]: I'd not considered this; is this something you want to veto the feature with, or that just needs including in the comment? Maybe this should be a config daemon.conf option, default to allowing?

[the remark this answers is not in the document]
Comment (superm1) [reply libfwupdplugin/fu-device.h:455]: I think a config option is pragmatic.

[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-device.h:455]: Alternate proposal in https://github.com/fwupd/fwupd/pull/4458

**Pull request closed without merging**